game/src/game/services/GameStateService.h

```diff
@@ -70,6 +70,23 @@ class GameStateService : public Service, public IEventSubscriber<OnGuiEvent>
 
     void RegisterCheckpoint(Entity& entity, Checkpoint* checkpoint);
 
+    std::map<uint32_t, PlayerStateData> player_details_;
+    std::map<uint32_t, PlayerState*> player_states_;
+    std::map<uint32_t, PowerupPickupType> player_powers_;
+    std::set<std::pair<uint32_t, PowerupPickupType>> same_powerup_;
+
+    std::vector<std::pair<uint32_t, PowerupPickupType>> active_powerups_;
+    std::map<std::pair<uint32_t, PowerupPickupType>, float> timer_;
+
+    GameState stats_;
+
+    void StartCountdown();
+    void StartGame();
```

Comment:
Can these be moved back down into the private section?

Also, can these be changed back to `std::unordered_map` instead? `std::map` is sorted by the key by default, and we don't need the sorting